src/transformers/hf_argparser.py

```diff
@@ -65,6 +65,14 @@ def _add_dataclass_arguments(self, dtype: DataClassType):
                 if field.default is True:
                     field_name = f"--no-{field.name}"
                     kwargs["dest"] = field.name
+            elif hasattr(field.type, "__origin__") and issubclass(field.type.__origin__, List):
```

Comment:
@julien-c - I need that for `list` inputs to the `HfArgParser`.

Reply:
LGTM, but can you also add the corresponding tests to `test_hf_argparser.py`?

Reply:
> Should `List` also allow `Tuple`?

I think it's hard to parse a list like object from the command line like `--tuple_arg 1 2 3`  with `argparse.parse` that will not come out as a list but as a tuple. Also I don't really see the need to be able to parse tuples if one can parse lists in data classes. It would be easy to allow tuples in dataclasses to be parsed via the command line as a list, but then why would one want to parse something into a tuple and not into  a list... => so I decided to just stick to list for now